### PR TITLE
issue 1645

### PR DESCRIPTION
**Pull request by vporton**
successfully tested

### CODE REVIEW

File: test/issue1645-dflags-build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+# If DFLAGS are not processed, dub for library would fail
+DFLAGS="-w" $DUB --root="$CURR_DIR"/1-staticLib-simple --build=plain
+if DFLAGS="-asfdsf" $DUB --root="$CURR_DIR"/1-staticLib-simple --build=plain 2>/dev/null; then
+  echo "Should not accept this DFLAGS"
+fi
```
Comment (wilzbach) [inline test/issue1645-dflags-build.sh]: This isn't testing anything. It will always pass. You need to use at least `set -e`

Comment (vporton) [reply test/issue1645-dflags-build.sh]: Fixed.

**Comment by wilzbach** [comment]
@vporton this still fails on the CIs.

```
[INFO] Running /home/travis/build/dlang/dub/test/issue1645-dflags-build.sh...
Performing "plain" build using dmd for x86_64.
staticlib-simple ~master: building configuration "library"...
Target is a library. Skipping execution.
[ERROR] Script failure.
```

**Comment by joseph-wakeling-frequenz** [comment]
Have I understood right that the intended (and effective) behaviour of this change is that command-line set `DFLAGS` will be taken _in addition_ to anything set by the `--build` flag?  And that one must explicitly specify `--build="$DFLAGS"` to use just the environment-set DFLAGS?

If so (i) I like this, because the logical use of `DFLAGS='...' dub build ...` is to add extra custom flags, but (ii) it might be worth tweaking the docs for `dub build` to explain this.

**Comment by Geod24** [comment]
Fixed in #2169

**Pull request closed without merging**